libgnucash/engine/gnc-date.cpp

```diff
@@ -126,61 +126,55 @@ gnc_localtime_r (const time64 *secs, struct tm* time)
 }
 
 static void
-normalize_time_component (int *inner, int *outer, unsigned int divisor,
-                          int base)
+normalize_time_component (int *inner, int *outer, int divisor, int base)
 {
-     while (*inner < base)
-     {
-          --(*outer);
-          *inner += divisor;
-     }
-     while (*inner > static_cast<gint>(divisor))
-     {
-          ++(*outer);
-          *inner -= divisor;
-     }
+    if (*inner < base)
+    {
+        do
+        {
+            --(*outer);
+            *inner += divisor;
+        } while (*inner < base) ;
+        return;
+    }
```

Comment:
I'm less keen on the `if`-`do`-`while`-`return` construct here. It looks clumsier and [doesn't save a compare, just moves it](https://godbolt.org/z/d463q4ad3).

Reply:
The `return` does save a compare.

Reply:
It doesn't. It just trades it for the `if (*inner  < base)`.

Reply:
It does.  The `return` saves the `*inner >= divisor + base` .